lib/r10k/module/metadata_file.rb

```diff
@@ -0,0 +1,31 @@
+require 'r10k/module'
+require 'shared/puppet/module_tool/metadata'
+
+class R10K::Module::MetadataFile
+
+  # @param metadata_path [Pathname] The file path to the metadata
+  def initialize(metadata_file_path)
+    @metadata_file_path = metadata_file_path
+  end
+
+  # Does the metadata file itself exist?
+  def exist?
+    @metadata_file_path.file? and @metadata_file_path.readable?
+  end
+
+  # Attempt to read the metadata file and return the Puppet::ModuleTool::Metadata object
+  def read(metadata_file_path = @metadata_file_path)
+    if self.exist?
+      File.open(metadata_file_path) do |f|
+        begin
+          metadata = Puppet::ModuleTool::Metadata.new
+          #returns Metadata.self
+          metadata.update(JSON.load(f))
+        rescue JSON::ParserError
+          #raise ArgumentError, "Could not parse JSON #{metadata_path}", ex.backtrace
```

Comment:
This is swallowing an exception, so it might be good to log a message at debug or debug1 to indicate that the load failed.